air/tests/test_module/instructions/fail.rs

```diff
@@ -97,6 +124,34 @@ fn fail_with_last_error_tetraplets() {
     );
 }
 
+#[test]
+fn fail_with_error_tetraplets() {
+    let local_peer_id = "local_peer_id";
+    let fallible_service_id = "service_id_1";
+    let (host_closure, tetraplet_anchor) = tetraplet_host_function(fallible_call_service(fallible_service_id));
+    let mut vm = create_avm(host_closure, local_peer_id);
+
+    let local_fn_name = "local_fn_name";
+    let script = format!(
+        r#"
+        (xor
+            (xor
+                (call "{local_peer_id}" ("{fallible_service_id}" "{local_fn_name}") [] result_1)
+                (fail :error:)
```

Comment:
```suggestion
                (seq
                  (ap :error: error_object_scalar)
                  (fail :error:)
                )
```

Reply:
But it will check value of `:error:` before the `fail`, not after that.
I don't see how to do it correctly, though.